### PR TITLE
add is_cancelled() to build_response.py

### PR DESCRIPTION
**Pull request by maxamillion**
Signed-off-by: Adam Miller <maxamillion@fedoraproject.org>

### CODE REVIEW

**Comment by maxamillion** [comment]
@twaugh this doesn't change state. Also, I couldn't find any tests for any of the `BuildResponse` "status checkers" (`is_failed`, `is_running`, `is_pending`, etc). Should I just write new tests and go ahead and cover all of those for good measure?

**Comment by twaugh** [comment]
We have an existing test for cancelling a build -- it could check what this method says before and after.

**Comment by maxamillion** [comment]
@twaugh I see what you mean, I'll get this added. Thanks

File: tests/build_/test_build_response.py [hunk 1]
```diff
@@ -56,6 +56,31 @@ def test_build_cancel(self):
...
+    def test_state_checkers(self):
+        build_response = BuildResponse({
+            'status': {
+                'phase': 'Complete'
+            }
+        })
+
+        build_response.status = 'complete'
+        assert build_response.is_finished()
+        assert build_response.is_succeeded()
```
Comment (twaugh) [inline tests/build_/test_build_response.py]: Could also `assert not build_response.is_failed()` etc.

Comment (maxamillion) [inline tests/build_/test_build_response.py]: +1 will do

**Comment by twaugh** [comment]
Would be nice to squash commits I think.

**Comment by maxamillion** [comment]
All squashed, thanks all!

**Comment by twaugh** [comment]
Draft release notes for 0.41 updated.

**Pull request merged**